src/main/flight/pid.c

```diff
@@ -253,21 +253,77 @@ void pidInitConfig(const pidProfile_t *pidProfile) {
     relaxFactor = 1.0f / (pidProfile->setpointRelaxRatio / 100.0f);
     levelGain = pidProfile->P8[PIDLEVEL] / 10.0f;
     horizonGain = pidProfile->I8[PIDLEVEL] / 10.0f;
-    horizonTransition = 100.0f / pidProfile->D8[PIDLEVEL];
+    horizonTransition = (float)pidProfile->D8[PIDLEVEL];
```

Comment:
Well see my other comment. It is because you dont initialise other parameters too.

Your code fits really well in the old "style" pid code where there was tons of rescaling of configurator values. We kind of got rid of all that, which makes reading of the pid code much easier. It would be nice if you could rework your code to initialised part and active part like the rest